Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -507,10 +508,56 @@ public bool ClearStateCache(VBComponent component)
                 Debug.WriteLine(string.Format("Removed Project declaration for project Id {0}", projectId));
             }
 
+            if (setStateChanged)
+            {
+                OnStateChanged();
+            }
+
             Debug.WriteLine("ClearDeclarations({0}): {1} - {2} declarations removed", component.Name, success ? "succeeded" : "failed", declarationsRemoved);
             return success;
         }
 
+        public bool RemoveRenamedComponent(VBComponent component, string oldComponentName)
```

Comment:
I don't like the copy-paste going on here, this isn't trivial logic, and it's too easy to modify one place and miss the other.